### PR TITLE
Fix Solaris 11 Builds

### PR DESCRIPTION
**Pull request by ScottGarman**
A while back I had mistakenly excluded solaris builds from the cross_compiled check which includes the platform architecture in the package name. This reverts that commit and fixes errors when building the pl-boost package for solaris-11-i386.

### CODE REVIEW

File: configs/projects/pl-boost.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@
   proj.vendor "Puppet Labs <info@puppetlabs.com>"
   proj.homepage "https://www.puppetlabs.com"
 
-  if platform.is_cross_compiled_linux?
+  if platform.is_cross_compiled? || platform.name =~ /solaris-11/
```
Comment (MikaelSmith) [inline configs/projects/pl-boost.rb]: Why is this needed on Solaris 11?

Comment (ScottGarman) [reply configs/projects/pl-boost.rb]: It's due to this in puppet-agent: https://github.com/puppetlabs/puppet-agent/blob/78ff7d8bd6bf5f2e78f3f2e26255bde0238053b9/configs/components/leatherman.rb#L14

I have a ticket (BUILD-107) to clean this up. My goal with this PR was just to get things working with the p-a repo as it was.

**Comment by McdonaldSeanp** [comment]
👍
